File: filter-pattern/python/src/CriteriaMale.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""
from src.Criteria import Criteria


# 根据标准接口实现的过滤
class CriteriaMale(Criteria):

    def filter(self, persons):
        male_persons = []
        for person in persons:
            if person.get_gender().upper() == 'MALE':
                male_persons.append(person)

        return male_persons

File: filter-pattern/ts/src/CriteriaMale.js
export class CriteriaMale {
    filter(persons) {
        const malePersons = new Array();
        for (const person of persons) {
            if (person.getGender().toUpperCase() === 'MALE') {
                malePersons.push(person);
            }
        }
        return malePersons;
    }
}
